frame/election-provider-multi-phase/src/lib.rs

```diff
@@ -1570,6 +1577,80 @@ impl<T: Config> Pallet<T> {
 	}
 }
 
+#[cfg(any(feature = "try-runtime", test))]
+impl<T: Config> Pallet<T> {
+	fn do_try_state() -> Result<(), &'static str> {
+		Self::try_state_snapshot()?;
+		Self::try_state_signed_submissions_map()?;
+		Self::try_state_phase_off()
+	}
+
+	// [`Snapshot`] state check. Invariants:
+	// - [`DesiredTargets`] exist IFF [`Snapshot`] is present.
+	// - [`SnapshotMetadata`] exist IFF [`Snapshot`] is present.
+	fn try_state_snapshot() -> Result<(), &'static str> {
+		let set = <DesiredTargets<T>>::get().is_some() && <SnapshotMetadata<T>>::get().is_some();
```

Comment:
>  [`DesiredTargets`] exist IFF [`Snapshot`] is present.

If `DesiredTargets` exists but `SnapshotMetadata` and `Snapshot` does not exist, the current implementation will return no error.

Reply:
May be a better and simpler assertion to make is 
```
ensure!(<DesiredTargets<T>>::get().is_some() == <SnapshotMetadata<T>>::get().is_some() && <DesiredTargets<T>>::get().is_some() == <Snapshot<T>>::get().is_some(), "snapshot invariant expectations not met");
```

Reply:
> If DesiredTargets exists but SnapshotMetadata and Snapshot does not exist, the current implementation will return no error.

That is the expected behaviour. One of the invariants is that `DesiredTargets` is set only if `Snapshot` is set. Another invariant is that `SnapshotMetadata` should be set only is `Snapshot` is set.

I will replace `IFF` in the comments with "if and only if", it may be easier to grok.

Reply:
If `Snapshot` is not set, but `DesiredTargets` is set, we expect the try-state checks to throw an error, correct?

But it won't throw an error if in addition to above conditions, `SnapshotMetadata` also does not exist.

To clarify:
Given: 
`DesiredTarget` exists.
`SnapshotMetadata` does not exist.
`Snapshot` does not exist.

set = true && false => false
Snapshot is `None` and `!set` which goes to catchall arm and returns Ok().

Reply:
I would write this as `ensure!(snapshot::exists() && metadata::exists() && desiredTargets::exists())`

These 3 storage items should always live together, and be destroyed together, which makes the a good candidate to be privatized similar to https://github.com/paritytech/substrate/pull/13950#issuecomment-1515959886.